packages/remix-react/browser.tsx

```diff
@@ -198,7 +200,10 @@ export function RemixBrowser(_props: RemixBrowserProps): ReactElement {
     // towards determining the route matches.
     let initialPathname = window.__remixContext.url;
     let hydratedPathname = window.location.pathname;
-    if (initialPathname !== hydratedPathname) {
+    if (
+      initialPathname !== hydratedPathname &&
+      !window.__remixContext.isSpaMode
+    ) {
```

Comment:
Don't perform this check in SPA mode since we want to be able to hydrate any URL via `route.lazy`